### PR TITLE
Homework done.

### PR DESCRIPTION
**Pull request by Koczek9**
Fixed typo in Prime number test, created sumNumbersFromGenerators function.

### CODE REVIEW

File: src/NumberGenerator.hpp [hunk 1]
```diff
@@ -8,6 +8,7 @@ enum class GeneratorType
 
 struct NumberGenerator
 {
+    virtual ~NumberGenerator();
```
Comment (ziobron) [inline src/NumberGenerator.hpp]: You are missing the implementation of this virtual destructor.
```suggestion
    virtual ~NumberGenerator(){}
```

**Review by ziobron — CHANGES_REQUESTED** [review]
Your solution is correct, but it does not compile because of linker error - https://travis-ci.com/github/coders-school/pre_work/jobs/360423321
It's nice that you have also found a hidden problem with missing virtual destructor.

File: src/NumberGenerator.hpp (hunk 1 above)
Comment (Koczek9) [reply src/NumberGenerator.hpp]: Yes, I was just pushing this change :)

**Review by ziobron — APPROVED** [review]
👍

**Pull request closed without merging**